src/main/scala/dpla/ingestion3/enrichments/SpatialEnrichment.scala

```diff
@@ -0,0 +1,275 @@
+    *
+    * @see enrich()
+    * @param j  The JValue object for the Twofishes response body
+    * @return   The woeType
+    */
+  private def woeType(j: JValue): Option[Int] = {
+    Option(((j \ "interpretations")(0) \ "feature")
+      .extract[Feature]
+      .woeType)
+  }
+
+  /**
+    * Return a parent feature's name, given the Twofishes response and a WoeType
+    * (What-on-Earth Type) for the rank of the feature (city, state, etc.).
+    *
+    * @param json           Twofishes response body JValue
+    * @param filterWoeType  What-on-Earth Type number
+    * @return
+    */
+  private def parentFeatureName(json: JValue, filterWoeType: Int):
+      Option[String] = {
+    val featNames: List[String] = for {
+      JArray(parents) <- json \\ "parents"
+      JObject(parent) <- parents
+      JField("name", JString(name)) <- parent
+      JField("woeType", JInt(woeType)) <- parent
+      if woeType == filterWoeType
+    } yield name
+    if (featNames.nonEmpty) Option(featNames.head)
+    else None
+  }
+
+  /**
+    * Return the city / town name of the Place.
+    *
+    * Note that this is not handled by `parentFeatureName` because the City name is
+    * going to be the name of the place, City / Town being the smallest level
+    * of detail reported by Twofishes. It's not going to be a parent feature.
+    *
+    * @param j  Twofishes response body JValue
+    * @return   The city / town name
+    */
+  private def city(j: JValue): Option[String] = {
+    val feature = ((j \ "interpretations")(0) \ "feature").extract[Feature]
+    if (feature.woeType == 7) {
+      Option(feature.name)
+    } else {
+      None
+    }
+  }
+
+  /* Similar to `city`, return the other parent feature names ... */
+  private def county(j: JValue): Option[String] = {
+    if (woeType(j) == Some(9)) shortName(j)
+    else parentFeatureName(j, 9)
+  }
+  private def state(j: JValue): Option[String] = {
+    if (woeType(j) == Some(8)) shortName(j)
+    else parentFeatureName(j, 8)
+  }
+  private def country(j: JValue): Option[String] = {
+    if (woeType(j) == Some(12)) shortName(j)
+    else parentFeatureName(j, 12)
+  }
+
+  // Region is implemented inconsistently in the Ingestion1 Twofishes enricher,
+  // and Twofishes does not return it consistently. If you look up "United
+  // States" it will provide the continent, "North America", but if you look up
+  // a city in the United States it will not return the continent.  Filling in
+  // a region value will provide misleading searches.
+  // private def region(j: JValue): String = ???
+
+  /**
+    * Return a single String of the Place's coordinates, joined by ","
+    *
+    * @param j  Twofishes response JValue
+    * @return   The coordinates as at "lat, lng" string
+    */
+  private def coordinates(j: JValue): Option[String] = {
+    val c = ((((j \ "interpretations")(0) \ "feature") \ "geometry") \
+      "center").extract[Coords]
+    Option(c.lat.toString + "," + c.lng.toString)
```

Comment:
Yeah, I looked it up. The Option(value) version is for if the value might be null, It returns None if null, Some(value) otherwise. So it promotes a type from a nullable type to a Option type that's not nullable.

Some(value) for something that's nullable will return Some(null) in those cases, which is usually not intended.